modules/core/src/main/java/org/apache/ignite/internal/visor/tx/VisorTxTask.java

```diff
@@ -321,6 +338,20 @@ else if (locTx instanceof GridDhtTxRemote) {
                     break;
             }
 
+            // If transaction was not found in verbose --tx --info mode, try to fetch it from history.
+            if (arg.txInfoArgument() != null && infos.isEmpty()) {
+                Map<GridCacheVersion, Object> completedVersions = U.field(tm, "completedVersHashMap");
```

Comment:
Why do we use hack here?
Could we replace by normal method?

Reply:
Fixed.